src/components/language/language.service.ts

```diff
@@ -684,8 +679,56 @@ export class LanguageService {
       items: result.items,
       hasMore: result.hasMore,
       total: result.total,
-      canCreate: !!readProject[0]?.canCreateProject,
-      canRead: !!readProject[0]?.canReadProjects,
+      //TODO: use the upcoming Roles service to determine permissions.
+      canCreate: true,
+      canRead: true,
+    };
+  }
+
+  async sponsorStartDate(
+    language: Language,
+    session: ISession
+  ): Promise<SecuredDate> {
+    const result = await this.db
+      .query()
+      .match([
+        node('', 'Language', { id: language.id }),
+        relation('in', '', 'language', { active: true }),
+        node('engagement', 'LanguageEngagement'),
+      ])
+      .return(collect('engagement.id', 'engagementIds'))
+      .asResult<{ engagementIds: string[] }>()
+      .first();
+
+    if (!result) {
+      throw new ServerException('Error fetching sponsorStartDate');
+    }
+
+    const engagments = await Promise.all(
+      result.engagementIds.map((engagementId) =>
+        this.engagementService.readOne(engagementId, session)
+      )
+    );
```

Comment:
I think readOne can throw NotFound exception. We should catch that and return canRead=false, etc response

Reply:
Added